webpack/assets/javascripts/react_app/components/Editor/Editor.js

```diff
@@ -147,16 +162,18 @@ class Editor extends React.Component {
           fetchAndPreview={fetchAndPreview}
         />
         <EditorView
-          key="editorView"
-          value={isRendering ? previewResult : value}
-          name={editorName}
-          mode={isRendering ? 'Text' : mode}
-          theme={theme}
-          keyBinding={keyBinding}
-          onChange={isRendering ? noop : changeEditorValue}
-          readOnly={readOnly || isRendering}
-          className={selectedView !== 'diff' ? 'ace_editor_form' : 'hidden'}
-          isMasked={isMasked}
+          {...editorViewProps}
+          key="editorPreview"
+          name={editorNameTab.preview}
+          isSelected={selectedView === 'preview'}
+          className="ace_editor_form ace_preview"
+        />
+        <EditorView
+          {...editorViewProps}
+          key="editorCode"
```

Comment:
maintaining two editors feels a bit like an hack,
maybe there is some easy way to keep track on the `element > scrollTop` of each tab when switching between tabs?

Reply:
It may actually be a good thing to do, I realized yesterday, ctrl+z works weird thanks to single editor, you can simply get back to preview/code/diff with this. We should either disable ctrl+z (sub optimal) or keep different editor for each tab.

Reply:
cc @glekner

Reply:
I think as these are 2 different text boxes they should be separated and not to switch the value of the same text box each time the user changes the tab

Reply:
Hey all, IMO having multiple instances of the editor could result in performance issues when going back and forth between the tabs. Maybe with `classname='hidden'` we could avoid the remounting of them, so you should check it performance wise.

I think resetting the scrolling state when moving between tabs is the ideal solution. but again if having two instances without hurting performance works then its also a valid solution.

Reply:
I tend to agree with @MariaAga and @ares, it looks like the scrolling and the ctrl+z are side effects due to the fact it's the same editor instance. Using multiple instances might fix other side effects we do not aware of and also prevent other side effects that might show up once we do changes to the editor.

Reply:
And it would be nice to measure if there are any performance differences.

Reply:
I checked the performance by adding a new template, previewing, editing again, and re-previewing:
1 editor:
4486 msScripting
494 msRendering
2 editors:
4500 msScripting
742 msRendering

Reply:
Thanks @MariaAga, IMO the difference is reasonable.